### PR TITLE
refactor: remove `onChangedNullable`, update `onChanged` for null val…

### PR DESCRIPTION
**Pull request by moshOntong-IT**
…ues, and set `allowDeselection` default to false

This commit introduces breaking changes:

1. Removed the `onChangedNullable` method to simplify the API and avoid developer confusion. The `onChanged` method now supports returning or callback with a null value. Developers must implement null checks in their code to handle nullable results from `onChanged`.

2. The `allowDeselection` property now defaults to `false`. Ensure your code explicitly sets this to `true` if deselection is needed. However, when the variant is multiple, the `allowDeselection` property defaults to `true`.

Please update your code to accommodate these changes.

### CODE REVIEW

File: example/lib/pages/select_form_field.dart [hunk 1]
```diff
@@ -81,10 +85,16 @@ class _SelectFormFieldPageState extends State<SelectFormFieldPage> {
               children: [
                 ShadSelectFormField(
                   id: 'email',
+                  allowDeselection: allowDeselection,
                   minWidth: 350,
                   initialValue: initialValue,
-                  onChanged: allowDeselection ? null : print,
-                  onChangedNullable: allowDeselection ? print : null,
+                  // onChanged: allowDeselection ? null : print,
```
Comment (nank1ro) [inline example/lib/pages/select_form_field.dart]: Remove the comments

File: lib/src/components/form/fields/select.dart [hunk 2]
```diff
@@ -65,19 +63,20 @@ class ShadSelectFormField<T> extends ShadFormBuilderField<T> {
...
               enabled: state.enabled,
-              onChanged: onChangedNullable != null ? state.didChange : null,
-              onChangedNullable: onChangedNullable != null
-                  ? (v) {
-                      state.didChange(v);
-                      onChangedNullable(v);
-                    }
-                  : null,
+              onChanged: onChanged != null ? state.didChange : null,
+              // onChangedNullable: onChangedNullable != null
```
Comment (nank1ro) [inline lib/src/components/form/fields/select.dart]: Remove comments

File: lib/src/components/form/fields/select.dart [hunk 3]
```diff
@@ -108,8 +107,8 @@ class ShadSelectFormField<T> extends ShadFormBuilderField<T> {
     super.description,
     super.onChanged,
 
-    /// {@macro ShadSelect.onChangedNullable}
-    ValueChanged<T?>? onChangedNullable,
+    // /// {@macro ShadSelect.onChangedNullable}
```
Comment (nank1ro) [inline lib/src/components/form/fields/select.dart]: Remove comments

File: lib/src/components/form/fields/select.dart [hunk 4]
```diff
@@ -162,19 +162,20 @@ class ShadSelectFormField<T> extends ShadFormBuilderField<T> {
...
-                      onChangedNullable(v);
-                    }
-                  : null,
+              onChanged: onChanged != null ? state.didChange : null,
+              // onChangedNullable: onChangedNullable != null
+              //     ? (v) {
+              //         state.didChange(v);
+              //         onChangedNullable(v);
+              //       }
+              //     : null,
```
Comment (nank1ro) [inline lib/src/components/form/fields/select.dart]: Remove comments

**Review by nank1ro — CHANGES_REQUESTED** [review]

File: example/lib/pages/select_form_field.dart (hunk 1 above)
Comment (moshOntong-IT) [reply example/lib/pages/select_form_field.dart]: done

File: lib/src/components/form/fields/select.dart (hunk 2 above)
Comment (moshOntong-IT) [reply lib/src/components/form/fields/select.dart]: done

File: lib/src/components/form/fields/select.dart (hunk 3 above)
Comment (moshOntong-IT) [reply lib/src/components/form/fields/select.dart]: done

File: lib/src/components/form/fields/select.dart (hunk 4 above)
Comment (moshOntong-IT) [reply lib/src/components/form/fields/select.dart]: Done

**Comment by moshOntong-IT** [comment]
I set the `allowDeselection` into true as a default when the variant type of Select is multiple

**Review by nank1ro — APPROVED** [review]
LGTM

**Pull request merged**